src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double MAX_FIGURE_SIZE = 11.0;
+    private static final double DEFAULT_RADIUS = 10.0;
+    private static final int NUMBER_OF_FIGURES = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(NUMBER_OF_FIGURES);
+
+        switch (figureType) {
+            case 0:
+                return createNewSquare();
+            case 1:
+                return createNewRectangle();
+            case 2:
+                return createNewRightTriangle();
+            case 3:
+                return createNewCircle();
+            case 4:
+                return createNewIsoscelesTrapezoid();
+            default:
+                return getDefaultFigure();
+        }
+
+    }
+
+    public Figure createNewSquare() {
+        double randomValue = random.nextDouble() * MAX_FIGURE_SIZE;

```

Comment:
Create separate method for getting randomg side and use it for each type of figure

random.nextDouble() * MAX_FIGURE_SIZE